packages/providers/src/provider.ts

```diff
@@ -603,6 +603,22 @@ export default class Provider {
     return contract;
   }
 
+  /**
+   * Returns the balance for the given contract for the given asset ID
+   */
+  async getContractBalance(
+    /** The contract ID to get the balance for */
+    contractId: AbstractAddress,
+    /** The asset ID of coins to get */
+    assetId: BytesLike
+  ): Promise<BN> {
+    const { contractBalance } = await this.operations.getContractBalance({
+      contract: contractId.toB256(),
+      asset: hexlify(assetId),
+    });
+    return bn(contractBalance.amount, 10);
```

Comment:
> defaults to bn(amount, hex). No need for the 10

Do `bn(amount, 'hex')` and `bn(amount, 10)` evaluate to the same result? Shouldn't `bn(amount, 16)` evaluate to the same as `'hex'`?

Reply:
Yeah, this is the question if we check fi the value was correct if it's than we have some issues on the file because some of the bn instances are not passing 10.

Reply:
Ok I check the value from the playground it returns a base 10 not a hex:

```
{
  "data": {
    "contractBalance": {
      "amount": "1182963097192509693",
      "assetId": "0x6dfc0f524d3006e103fc4bd733c4f012b5a089fd09361d44833cdf653e9afe5b"
    }
  }
}
```